src/azure-cli-testsdk/azure/cli/testsdk/scenario_tests/decorators.py

```diff
@@ -16,10 +16,13 @@ def live_only():
         'This is a live only test. A live test will bypass all vcrpy components.')
 
 
-def record_only():
+def record_only(keep_recording_file=False):
+    if keep_recording_file:
+        os.environ[ENV_KEEP_RECORDING_FILE] = 'True'
```

Comment:
If a previous test function set this environment True, when will it be reset as false? If it's not reset, it may pollute the following tests.